Add WEMOS LOLIN S3 Mini Pro board

sillyfrog opened the pull request:
## Description:

Adds the WEMOS LOLIN S3 Mini Pro board. Vendor board here: https://www.wemos.cc/en/latest/s3/s3_mini_pro.html

## Checklist:
  - [X] The pull request is done against the latest develop branch
  - [X] Only relevant files were touched
  - [X] Only one feature/fix was added per PR, more changes are allowed when changing boards.json
  - [X] I accept the [CLA](https://github.com/pioarduino/platform-espressif32/blob/main/CONTRIBUTING.md#contributor-license-agreement-cla)

sivar2311 commented on boards/lolin_s3_mini_pro.json:
```diff
@@ -0,0 +1,47 @@
+      "extra_flags": [
+        "-DBOARD_HAS_PSRAM",
+        "-DARDUINO_LOLIN_S3_MINI",
+        "-DARDUINO_USB_MODE=1"
```
The S3 can run in `USB_MODE = 1` **or** `USB_MODE = 0`. Therfore `-D ARDUINO_USB_MODE=1` should be removed and set on demand in the project's `platformio.ini`. 

Since there is no USB-2-Serial Chip, the extra_flag `-D ARDUINO_USB_CDC_ON_BOOT` should be set to `1`. This will "route" all `Serial.print...` to the correct port.

sivar2311 commented on boards/lolin_s3_mini_pro.json:
```diff
@@ -0,0 +1,47 @@
+      ],
+      "f_cpu": "240000000L",
+      "f_flash": "80000000L",
+      "flash_mode": "qio",
```
I'm missing the  `"psram_type" : "qio",` here.

sivar2311 commented on boards/lolin_s3_mini_pro.json:
```diff
@@ -0,0 +1,47 @@
+{
+    "build": {
+      "arduino": {
+        "ldscript": "esp32s3_out.ld",
```
`"ldscript" ... ` is outdated and should be removed

Jason2866 commented:
Please do, when you are in editing mode ;-)

sillyfrog commented:
Thanks for the quick responses! I've updated (and tested) the updated configs (and removed the previously required `build_flags` from my `platformio.ini`) for the _Wemos S3 Mini_ and _Wemos S3 Mini Pro_.
